test/integration/filters/test_listener_filter.cc

```diff
@@ -0,0 +1,34 @@
+#include "test/integration/filters/test_listener_filter.h"
+
+namespace Envoy {
+
+/**
+ * Config registration for the test filter.
+ */
+class TestInspectorConfigFactory : public Server::Configuration::NamedListenerFilterConfigFactory {
+public:
+  // NamedListenerFilterConfigFactory
+  Network::ListenerFilterFactoryCb createListenerFilterFactoryFromProto(
+      const Protobuf::Message& /*message*/,
+      const Network::ListenerFilterMatcherSharedPtr& listener_filter_matcher,
+      Server::Configuration::ListenerFactoryContext& /*context*/) override {
+    return [listener_filter_matcher](Network::ListenerFilterManager& filter_manager) -> void {
+      filter_manager.addAcceptFilter(listener_filter_matcher,
+                                     std::make_unique<TestListenerFilter>());
+    };
+  }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Struct()};
+  }
+
+  std::string name() const override { return "envoy.filters.listener.test"; }
+};
+
+absl::Mutex TestListenerFilter::alpn_lock_;
```

Comment:
Another way is to set the alpn as a parameter for this filter config, then we can get per-instance alpn value. But yes, it needs more code.

Reply:
I think that doesn't work as we want one filter which returns different alpn values for different connections.  But anyway the goal is to just have those tests pass without a filter some folks don't want to include so I think simple (with crummy locks) is fine :-)